client/state/plugins/installed/test/selectors-ts.ts

```diff
@@ -213,3 +217,51 @@ describe( 'getAllPluginsIndexedBySiteId', () => {
 		expect( pluginsOne ).toBe( pluginsTwo );
 	} );
 } );
+
+describe( 'getFilteredAndSortedPlugins', () => {
+	test( 'Should get an empty array if the requested site is not in the current state', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ nonExistingSiteId ], undefined );
+		expect( plugins ).toHaveLength( 0 );
+	} );
+
+	test( 'Should get an empty array if the plugins for this site are still being requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteThreeId ], undefined );
+		expect( plugins ).toHaveLength( 0 );
+	} );
+
+	test( 'Should get a plugin list of length 3 if both sites are requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], undefined );
+		expect( plugins ).toHaveLength( 3 );
+	} );
+
+	test( 'Should get a plugin list containing jetpack if both sites are requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], undefined );
+		const siteWithPlugin = {
+			[ siteTwoId ]: {
+				active: jetpack.active,
+				autoupdate: jetpack.autoupdate,
+				update: jetpack.update,
+				version: jetpack.version,
+			},
+		};
```

Comment:
Nit: this could be simplified by using destructuring:

```suggestion
		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], undefined );
		const { active, autoupdate, update, version } = jetpack;
		const siteWithPlugin = {
			[ siteTwoId ]: { active, autoupdate, update, version },
		};
```